### ISSUE TITLE
Failed build on i686-w64-mingw32 

### ISSUE THREAD

**Issue by caobug** [warmcat/libwebsockets#1939]
Hello, this latest version 4.0.13 can't build on Windows

```
$ gcc -v
Using built-in specs.
COLLECT_GCC=C:\msys64\mingw32\bin\gcc.exe
COLLECT_LTO_WRAPPER=C:/msys64/mingw32/bin/../lib/gcc/i686-w64-mingw32/10.1.0/lto-wrapper.exe
Target: i686-w64-mingw32
Configured with: ../gcc-10.1.0/configure --prefix=/mingw32 --with-local-prefix=/mingw32/local --build=i686-w64-mingw32 --host=i686-w64-mingw32 --target=i686-w64-mingw32 --with-native-system-header-dir=/mingw32/i686-w64-mingw32/include --libexecdir=/mingw32/lib --enable-bootstrap --with-arch=i686 --with-tune=generic --enable-languages=c,lto,c++,fortran,ada,objc,obj-c++ --enable-shared --enable-static --enable-libatomic --enable-threads=posix --enable-graphite --enable-fully-dynamic-string --enable-libstdcxx-filesystem-ts=yes --enable-libstdcxx-time=yes --disable-libstdcxx-pch --disable-libstdcxx-debug --disable-isl-version-check --enable-lto --enable-libgomp --disable-multilib --enable-checking=release --disable-rpath --disable-win32-registry --disable-nls --disable-werror --disable-symvers --disable-plugin --with-libiconv --with-system-zlib --with-gmp=/mingw32 --with-mpfr=/mingw32 --with-mpc=/mingw32 --with-isl=/mingw32 --with-pkgversion='Rev2, Built by MSYS2 project' --with-bugurl=https://sourceforge.net/projects/msys2 --with-gnu-as --with-gnu-ld --disable-sjlj-exceptions --with-dwarf2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 10.1.0 (Rev2, Built by MSYS2 project)
```

```
libwebsockets-4.0.13\lib\core-net\output.c:369:19: error: cast from pointer to integer of different size [-Werror=pointer-to-int-cast]
  369 |    n = write((int)(long long)wsi->desc.filefd, buf, len);
      |              ^
```

I added `-DCMAKE_C_FLAGS="-Wno-pointer-to-int-cast"` and rebuild, is ok.

============================

Old version 3.2.3 can successfully  build, however if I include `libwebsockets.h` in my project,  other errors will appear:

```
multiple definition of `humanize_schema_si';
multiple definition of `humanize_schema_si_bytes';
multiple definition of `humanize_schema_us';
```

**Comment by lws-team** [comment]
>  4.0.13 can't build on Windows

Hm, it can build on windows just fine using msvc

https://ci.appveyor.com/project/lws-team/libwebsockets/builds/33243622

> (int)(long long)wsi->

Anything like this needs upgrading to (int)(lws_intptr_t)... I thought they were all done because msvc stopped complaining, I'll provide a patch later.

> 3.2.3 humanize

master, v4.0-stable and v3.2.3 all provide exports for that like this

LWS_VISIBLE LWS_EXTERN const lws_humanize_unit_t humanize_schema_si[7];

**Comment by caobug** [comment]
Thanks.

**Comment by lws-team** [comment]
I pushed a patch on v4.0-stable and master for the lws_intptr_t bits.

And on master I added mingw32 and 64 support in lws CI, since Fedora has packages for it and its cross root.

I think the problem with humanize_ is normally we use LWS_EXTERN in the api exports, IIRC for windows it is defined to be empty.  Does the problem with that go away if you replace the related LWS_EXTERN in lws-misc.h for the data exports with `extern`?
